src/main/drivers/serial_uart_stm32f10x.c

```diff
@@ -37,22 +38,158 @@
 #include "serial_uart.h"
 #include "serial_uart_impl.h"
 
+typedef enum UARTDevice {
+    UARTDEV_1 = 0,
+    UARTDEV_2 = 1,
+    UARTDEV_3 = 2,
+    UARTDEV_4 = 3,
+    UARTDEV_5 = 4,
+    UARTDEV_6 = 5
+} UARTDevice;
+
+typedef struct uartPinPair_s {
+    ioTag_t rx;
+    ioTag_t tx;
+} uartPinPair_t;
+
+#define UART_PINPAIR_COUNT 3
+
+typedef struct uartDevice_s {
+    USART_TypeDef* dev;
+    uartPort_t port;
+    DMA_Channel_TypeDef *txDMAChannel;
+    DMA_Channel_TypeDef *rxDMAChannel;
+    uartPinPair_t pinPair[UART_PINPAIR_COUNT];
+    ioTag_t rx;
+    ioTag_t tx;
+    volatile uint8_t *rxBuffer;
+    volatile uint8_t *txBuffer;
+    rccPeriphTag_t rcc;
+    uint8_t af;
+    uint8_t irqn;
+    uint32_t txPriority;
+    uint32_t rxPriority;
+} uartDevice_t;
+
 #ifdef USE_UART1
-static uartPort_t uartPort1;
+//static uartPort_t uartPort1;
 #endif
 
 #ifdef USE_UART2
-static uartPort_t uartPort2;
+//static uartPort_t uartPort2;
 #endif
 
 #ifdef USE_UART3
 static uartPort_t uartPort3;
 #endif
 
+// When using RX DMA, TX DMA must be enabled.
```

Comment:
Yes, there is (was).

`if` for UART interrupt was only checking if RX DMA is not used, causing the interrupt not enabled for RX DMA=ON, TX DMA=OFF case, which caused TX side inoperable.